src/Mod/AddonManager/addonmanager_utilities.py

```diff
@@ -296,21 +286,50 @@ def getDescRegex(url):
     """Returns a regex string that extracts a WB description to be displayed in the description
     panel of the Addon manager, if the README could not be found"""
 
-    if ("github" in url):
+    if "github" in url:
         return "<meta property=\"og:description\" content=\"(.*?)\""
-    elif ("framagit" in url):
+    elif "framagit" in url or "gitlab" in url:
         return "<meta.*?content=\"(.*?)\".*?og\:description.*?>"
     print("Debug: addonmanager_utilities.getDescRegex: Unknown git host:",url)
     return None
 
-def getRepoUrl(text):
+
+def getReadmeHTMLUrl(url):
+    
+    "Returns the location of a html file containing readme"
+
+    if ("github" in url):
+        return url+"/blob/master/README.md"
+    else:
+        print("Debug: addonmanager_utilities.getReadmeUrl: Unknown git host:",url)
+        return None
+
+
+def getReadmeRegex(url):
     
-    "finds an URL in a given piece of text extracted from github's HTML"
+    """Return a regex string that extracts the contents to be displayed in the description
+    panel of the Addon manager, from raw HTML data (the readme's html rendering usually)"""
     
-    if ("href" in text):
-        return "https://github.com/" + re.findall("href=\"\/(.*?)\/tree",text)[0]
-    elif ("MOOC" in text):
-        # Bad hack for now... We need to do better
-        return "https://framagit.org/freecad-france/mooc-workbench"
-    print("Debug: addonmanager_utilities.getRepoUrl: Unable to find repo:",text)
-    return None
+    if ("github" in url):
+        return "<article.*?>(.*?)</article>"
+    else:
+        print("Debug: addonmanager_utilities.getReadmeRegex: Unknown git host:",url)
+        return None
+
+
+def fixRelativeLinks(text, base_url):
+
+    """Replace markdown image relative links with
+    absolute ones using the base URL"""
```

Comment:
No empty line before the docstring.

Reply:
Yes, it would be right thing to do. But this style is used all over this module.

Reply:
I could run [back](https://github.com/psf/black) on this module files after the review in separate commit.

Reply:
Please no, black is horrible. The style used in the file is bad, but we have better guidelines now, check Draft, Arch, Path, FEM workbenches. We know that older code doesn't quite comply with PEP8 but this is something we try to change little by little.

Reply:
I could do a code cleanup for this module in second commit